nengo/conftest.py

```diff
@@ -18,8 +19,7 @@
 
 
 def pytest_configure(config):
-    rc.reload_rc([])
-    rc.set('decoder_cache', 'enabled', 'False')
+    matplotlib.use('agg')
```

Comment:
Users having set their backend to something else is exactly the reason why I think that this is necessary. For example, on Linux my backend is set to 'Qt4Agg' right now, but `tox` won't install the required modules (e.g., pyside) and fail. On OS X, I have a similar problem where a framework install of Python is needed for the backend which is not supported by the virtualenv (without manual intervention).

Apart from that, the tests should produce reproducible results independent of what some user configuration file says (though, other matplotlib settings probably still have an effect). 'agg' should work on (almost?) all systems for the file output we're producing.